pkg/registry/core/service/allocator/bitmap.go

```diff
@@ -70,6 +71,50 @@ func NewAllocationMap(max int, rangeSpec string) *AllocationBitmap {
 		rangeSpec: rangeSpec,
 	}
 	return &a
+
+}
+
+// DynamicRangeOffset estimates the offset used on the range for statically allocation based on
+// the following formula `min(max($min, cidrSize/$step), $max)`, described as ~never less than
+// $min or more than $max, with a graduated step function between them~. The function returns 0
+// if any of the parameters is invalid.
+func DynamicRangeOffset(min, max, step, cidrSize int) int {
```

Comment:
I kind of feel like this belongs at the IP layer, rather than the bitmap layer.  It's very specific.  I think this layer should simply take a param which is the "preferred range size", which is avoided unless no other option.

Reply:
yeah, discussed in first commit. agree,